internal/model/vpnoptions.go

```diff
@@ -200,69 +203,9 @@ type TunnelInfo struct {
 	PeerID int
 }
 
-// NewTunnelInfoFromPushedOptions takes a map of string to array of strings, and returns
-// a new tunnel struct with the relevant info.
-func NewTunnelInfoFromPushedOptions(opts map[string][]string) *TunnelInfo {
-	t := &TunnelInfo{}
-	if r := opts["route"]; len(r) >= 1 {
-		t.GW = r[0]
-	} else if r := opts["route-gateway"]; len(r) >= 1 {
-		t.GW = r[0]
-	}
-	ifconfig := opts["ifconfig"]
-	if len(ifconfig) >= 1 {
-		t.IP = ifconfig[0]
-	}
-	if len(ifconfig) >= 2 {
-		t.NetMask = ifconfig[1]
-	}
-	peerID := opts["peer-id"]
-	if len(peerID) == 1 {
-		peer, err := strconv.Atoi(peerID[0])
-		if err != nil {
-			log.Println("Cannot parse peer-id:", err.Error())
-		} else {
-			t.PeerID = peer
-		}
-	}
-	return t
-}
-
-// parseIntFromOption parses an int from a null-terminated string
-func parseIntFromOption(s string) (int, error) {
-	str := ""
-	for i := 0; i < len(s); i++ {
-		if byte(s[i]) == 0x00 {
-			return strconv.Atoi(str)
-		}
-		str = str + string(s[i])
-	}
-	return 0, nil
-}
-
-// PushedOptionsAsMap returns a map for the server-pushed options,
-// where the options are the keys and each space-separated value is the value.
-// This function always returns an initialized map, even if empty.
-func PushedOptionsAsMap(pushedOptions []byte) map[string][]string {
-	optMap := make(map[string][]string)
-	if len(pushedOptions) == 0 {
-		return optMap
-	}
-
-	optStr := string(pushedOptions[:len(pushedOptions)-1])
-
-	opts := strings.Split(optStr, ",")
-	for _, opt := range opts {
-		vals := strings.Split(opt, " ")
-		k, v := vals[0], vals[1:]
-		optMap[k] = v
-	}
-	return optMap
-}
-
-func parseProto(p []string, o *OpenVPNOptions) error {
+func parseProto(p []string, o *OpenVPNOptions) (*OpenVPNOptions, error) {
```

Comment:
I'm missing some information to fully understand this change in pattern. AFAICT, it should not be necessary to return the `o` pointer, given that you're mutating it and it's available to the caller--can you elaborate?

Reply:
I suspect my reasoning here was to move to return a copy in the future and not modify the passed struct, but you're right that this might be unnecessary, especially since there's no other modification that uses this value.

do you think it's better to revert to the previous state and just return err?

Reply:
fwiw, I think mutating makes for confusing tests

```
617                         name:    "default",
618                         args:    args{o: &OpenVPNOptions{}},
619                         wantErr: nil,
```
and then instead of got := foo(tt.args.o} I need to check for mutation of the argument. I think it makes a cleaner API, but this needs more work than done here.